### PR TITLE
Chronos: support quantization for keras forecasters

### PR DESCRIPTION
**Pull request by plusbang**
## Description

Now, Nano upgrade INC to 2.0 and InferenceOptimizer also support custom model for keras INC.
Considering we only support quantization for pytorch forecasters, implement quantization for keras forecasters also.

### 2. User API changes

- a new API `quantization` for keras forecaster ()
```python
forecaster.quantize(input_data=train_data[0], target_data=train_data[1])
```
- a new parameter `quantize` in `predict` and `evaluate`
```python
def predict(self, data, batch_size=32, quantize=False)

def evaluate(self, data, batch_size=32, multioutput="raw_values", quantize=False)
```

### 3. Summary of the change 

- add `quantization`
- update `predict` and `evaluate` with a new parameter `quantize`
- enable TCN keras forecaster
- ut (**Note: capturing behaviors of `pytest -v` influence generating the valid sampling log during quantization. Work around by using `pytest -s` to test this ut.**)

### 4. How to test?
- [x] Unit test

### CODE REVIEW

**Review by TheaperDeng — COMMENTED** [review]
others LGTM, this should be a good first push and we could fix other problem once we have some feedback from users.

File: python/chronos/src/bigdl/chronos/forecaster/tf/base_forecaster.py [hunk 1]
```diff
@@ -109,8 +111,139 @@ def fit(self, data, epochs=1, batch_size=32):
                 self.internal.fit(data, epochs=epochs)
         self.fitted = True
 
-    def predict(self, data, batch_size=32):
+    def quantize(self, input_data=None,
+                 target_data=None,
+                 input_spec=None,
```
[the remark this answers is not in the document]
Comment (plusbang) [reply python/chronos/src/bigdl/chronos/forecaster/tf/base_forecaster.py:116]: > we don't need users to input this one right? we have the lookback and feature number and we could easily generate one for users.

yeah, we should generate it for users. I will remove this parameter `input_spec`. Besides, I think need to use `pytest -s` to test this ut in other actions also. I will add it in this PR.

**Review by TheaperDeng — APPROVED** [review]
LGTM, one question, the openvino and onnxruntime quantization model could not be called currently right?

**Comment by plusbang** [comment]
> LGTM, one question, the openvino and onnxruntime quantization model could not be called currently right?

Yes, I'm not sure whether we need to provide `predict_with_onnx` etc, `predict` only support accelerate_method=="tensorflow_int8" currently. We can work on it in later PR.

**Comment by TheaperDeng** [comment]
> > LGTM, one question, the openvino and onnxruntime quantization model could not be called currently right?
> 
> Yes, I'm not sure whether we need to provide `predict_with_onnx` etc, `predict` only support accelerate_method=="tensorflow_int8" currently. We can work on it in later PR.

Yeah I understand, we may leave this as a TODO for now.

**Pull request merged**